pallets/liquidity-proxy/src/lib.rs

```diff
@@ -549,16 +564,99 @@ impl<T: Config> Pallet<T> {
         )
     }
 
+    /// Exchange sequence of assets using output amount.
+    ///
+    /// Performs [`Self::exchange_single()`] for each pair of assets and aggregates the results.
+    fn exchange_sequence_with_output_amount(
+        dex_info: &DEXInfo<T::AssetId>,
+        sender: &T::AccountId,
+        receiver: &T::AccountId,
+        assets: &[(T::AssetId, T::AssetId, Balance, Balance)],
+        input_amount: SwapAmount<Balance>,
```

Comment:
You don't need this argument, because you already have all necessary data in `assets`